src/deadline/job_attachments/incremental_downloads/incremental_download_state.py

```diff
@@ -0,0 +1,185 @@
+        """
+        self.last_lookback_time = last_lookback_time
+        self.jobs = jobs or []
+
+    @classmethod
+    def from_dict(cls, data):
+        """
+        Create a IncrementalDownloadState instance from a dictionary.
+        Args:
+            data (dict): Dictionary containing state file data
+        Returns:
+            IncrementalDownloadState: A new instance populated with the data
+        """
+        if not data:
+            return cls()
+
+        return cls(last_lookback_time=data.get("lastLookbackTime"), jobs=data.get("jobs", []))
+
+    def to_dict(self):
+        """
+        Convert the IncrementalDownloadState to a dictionary.
+        Returns:
+            dict: Dictionary representation of the state file model
+        """
+        return {"lastLookbackTime": self.last_lookback_time, "jobs": self.jobs}
+
+    @classmethod
+    def from_bootstrap(
+        cls,
+        bootstrap_lookback_in_minutes: Optional[int],
+        print_function_callback: Callable[[str], None],
+    ) -> "IncrementalDownloadState":
+        """
+        Bootstraps fresh download progress state using bootstrap lookback if provided or now
+        :param bootstrap_lookback_in_minutes: lookback period in minutes
+        :param print_function_callback: Callback to print messages produced in this function.
+                Used in the CLI to print to stdout using click.echo. By default, ignores messages.
+        :return: returns a fresh download progress state
+        """
+        current_download_progress: IncrementalDownloadState = IncrementalDownloadState()
+
+        print_function_callback(
+            "Bootstrapping command. Ignoring download progress location and creating new"
+        )
+        current_download_progress.last_lookback_time = (
+            datetime.datetime.utcnow()
+            - datetime.timedelta(minutes=float(bootstrap_lookback_in_minutes or 0))
+        )
+        return current_download_progress
+
+    @classmethod
+    def from_file(
+        cls,
+        saved_progress_checkpoint_full_path: str,
+        print_function_callback: Callable[[str], None],
+    ) -> "IncrementalDownloadState":
+        """
+        Loads progress from state file saved at saved_progress_checkpoint_full_path
+        :param saved_progress_checkpoint_full_path: full path of the saved progress checkpoint file
+        :param print_function_callback: Callback to print messages produced in this function.
+                Used in the CLI to print to stdout using click.echo. By default, ignores messages.
+        :return: Returns the loaded state file,
+        or throws an exception if we're unable to read it as we already validated its existence
+        """
+        try:
+            with open(saved_progress_checkpoint_full_path, "r") as file:
+                state_data = json.load(file)
+                current_download_progress: IncrementalDownloadState = (
```

Comment:
nit: Good idea to de-dent from here down, the file doesn't need to stay open longer.